hs_core/hydroshare/resource.py

```diff
@@ -722,16 +725,32 @@ def delete_resource_file(pk, filename_or_id, user):
 
     return filename_or_id
 
+def get_resource_doi(res_id, pending):
+    doi_str = "http://dx.doi.org/10.4211/hs.{shortkey}".format(shortkey=res_id)
+    if pending:
+        return "{doi}pending".format(doi=doi_str)
+    else:
+        return doi_str
+
+
+def get_activated_doi(doi):
+    idx = doi.find('pending')
+    if idx >= 0:
+        return doi[:idx]
+    else:
+        return doi
 
-def publish_resource(pk):
+def publish_resource(user, pk):
```

Comment:
@dtarb @horsburgh I can try to do what you suggested to re-try metadata deposit with crossref periodically using a celery task, but before it is successful, what should we do with this pending resource? For other resources with publication metadata deposited successfully with crossref, we display a congratulations message initially and update the resources sharing status to published with citation data updated as well with DOI (with pending activation message before DOI is activated). What should we do with the resource in the interim before it is successfully deposited with crossref even with retried celery task?